src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -5,18 +5,28 @@
 using Stratis.Bitcoin.Base;
 using Stratis.Bitcoin.IntegrationTests.Common.EnvironmentMockUpHelpers;
 using Stratis.Bitcoin.P2P.Peer;
+using Xunit;
 
 namespace Stratis.Bitcoin.IntegrationTests.Common
 {
     public class TestHelper
     {
-        public static void WaitLoop(Func<bool> act)
+        public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reason", int millisecondsTimeout = 50, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var cancel = new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 30 * 1000);
+            cancellationToken = cancellationToken == default(CancellationToken)
+                ? new CancellationTokenSource(Debugger.IsAttached ? 15 * 60 * 1000 : 30 * 1000).Token
+                : cancellationToken;
             while (!act())
             {
-                cancel.Token.ThrowIfCancellationRequested();
-                Thread.Sleep(50);
+                try
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    Thread.Sleep(millisecondsTimeout);
```

Comment:
Can you elaborate a bit on what the benefit is of providing custom values for this?

Reply:
We gain ability to programatically control the sampling interval so that we can probe less or more frequently as well as we can increase the overall loop duration for processes that are likely to take longer then currently hardcoded 30 seconds. The thinking process behind this was that we want to maintain most checks to max 30[s] at 50[ms] interval, however we have ability to override defaults for edge cases. Other think to look into would be if the Thread being called with sleep method is appropriate here